app/uk/gov/hmrc/agentservicesaccount/auth/AuthActions.scala

```diff
@@ -47,20 +51,22 @@ class AuthActions @Inject()(logger: LoggerLike,
                             externalUrls: ExternalUrls,
                             override val authConnector: AuthConnector,
                             val env: Environment,
-                            val config: Configuration)(implicit val ec: ExecutionContext) extends AuthorisedFunctions with BaseController {
+                            val config: Configuration) extends AuthorisedFunctions with AuthRedirects {
 
-  def authorisedWithAgent(body: AgentInfo => Future[Result])(implicit request: Request[_]): Future[Result] =
+  def withAuthorisedAsAgent(body: AgentAuthAction)(implicit ec: ExecutionContext): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
any reason why EC is at method level now?

Reply:
Theres only one method in the class, does it make a difference?

Reply:
it's good if we have it at class level, who knows soon we may add more methods in this class, but its not strict for now, let me go through the remaining PR